### ISSUE TITLE
getMessages is not consistent

### ISSUE THREAD

**Issue by 0xktz** [gram-js/gramjs#158]
Hello, I am trying to fetch all the messages posted by a particular user in order to have them deleted afterwards.

This is my code:
`const res = await client.getMessages(channel, {
      limit: 500,
      fromUser: me
    })`

Where channel is of type PeerChannel and me is of type InputPeerUser.

My issue is that it doesn't seem to work consistently as occasionally it succeeds in fetching all the message Ids, but most of the time it returns just a single Id for each client which is the service message of the user joining the chat.

I tried various different arguments like for example setting minId:0, maxId: 99999999 or trying to set an offsetDate of Date.now() but without any actual luck.

Here is a debug trace of a successful fetch:

> [2021-09-19T04:34:43.952Z] [DEBUG] - [Assigned msgId = 7009498656511981176 to messages.Search]
[2021-09-19T04:34:43.952Z] [DEBUG] - [Encrypting 1 message(s) in 96 bytes for sending]
[2021-09-19T04:34:43.954Z] [DEBUG] - [Encrypted messages put in a queue to be sent]
[2021-09-19T04:34:43.955Z] [DEBUG] - [Waiting for messages to send...false]
[2021-09-19T04:34:43.955Z] [DEBUG] - [Assigned msgId = 7009498656523981260 to MsgsAck]
[2021-09-19T04:34:43.955Z] [DEBUG] - [Encrypting 1 message(s) in 68 bytes for sending]
[2021-09-19T04:34:43.956Z] [DEBUG] - [Encrypted messages put in a queue to be sent]
[2021-09-19T04:34:43.956Z] [DEBUG] - [Waiting for messages to send...false]
[2021-09-19T04:34:44.161Z] [DEBUG] - [Handling RPC result for message 7009498656511981176]
[2021-09-19T04:34:44.166Z] [DEBUG] - [Receiving items from the network...]

And the following is what occurs most of the times

> [2021-09-19T04:38:40.085Z] [DEBUG] - [Assigned msgId = 7009499670951230872 to messages.Search]
[2021-09-19T04:38:40.085Z] [DEBUG] - [Encrypting 1 message(s) in 96 bytes for sending]
[2021-09-19T04:38:40.087Z] [DEBUG] - [Encrypted messages put in a queue to be sent]
[2021-09-19T04:38:40.087Z] [DEBUG] - [Waiting for messages to send...false]
[2021-09-19T04:38:40.087Z] [DEBUG] - [Assigned msgId = 7009499670959230292 to MsgsAck]
[2021-09-19T04:38:40.087Z] [DEBUG] - [Encrypting 1 message(s) in 68 bytes for sending]
[2021-09-19T04:38:40.088Z] [DEBUG] - [Encrypted messages put in a queue to be sent]
[2021-09-19T04:38:40.088Z] [DEBUG] - [Waiting for messages to send...false]
[2021-09-19T04:38:40.301Z] [DEBUG] - [Handling RPC result for message 7009499670951230872]
[2021-09-19T04:38:40.302Z] [DEBUG] - [Receiving items from the network...]

Which looks virtually identical to the successful query.

Could this be related to any kind of throttling in place by the telegram API similar to GetHistory?
Also, is there any more efficient way for a user to delete his own messages without going through fetching their IDs first? I tried every method available without any actual luck sadly.

Thanks a lot!

**Comment by painor** [comment]
does this only happen when you use the from param?

alternatively of you are not the admin you can try using delete message with all ids and it will only delete the ones it can

**Comment by 0xktz** [comment]
Yes that is right, it works properly when the from parameter isn't used.

In regards to deletion I assume you refer to the deleteMessages method available to the TelegramClient, right? How would I go about inputing all IDs? Passing in an empty array doesn't do the trick, another thing that comes in mind is generating an array with all the possible IDs in the 32bit range and passing in this, is this what you have in mind?

**Comment by painor** [comment]
I wouldn't use that much but you can generate from 1 to current msgId something like `   Array.from({length:currentMsgId},(v,k)=>k+1)` and pass that to deleteMessages 
if you pass 10 000 ids it will use 10000/100 = 100 Api requests so you don't want to use a very big amount

**Comment by 0xktz** [comment]
Okay, thank you.

Using both of the workarounds suggested are enough to achieve my goal.
You may close the issue unless you'd like to have it open as for a reference to the original getMessages issue.
